tests/examples/inventory_management/inventory_env_pathmind.py

```diff
@@ -0,0 +1,404 @@
+        try:
+            assert self.num_periods > 0, "The number of periods must be positive. Num Periods = {}".format(self.num_periods)
+        except TypeError:
+            print('\n{}\n'.format(self.num_periods))
+        assert np.all(self.unit_price >= 0), "The sales prices cannot be negative."
+        assert np.all(self.unit_cost >= 0), "The procurement costs cannot be negative."
+        assert np.all(self.demand_cost >= 0), "The unfulfilled demand costs cannot be negative."
+        assert np.all(self.holding_cost >= 0), "The inventory holding costs cannot be negative."
+        assert np.all(self.supply_capacity > 0), "The supply capacities must be positive."
+        assert np.all(self.lead_time >= 0), "The lead times cannot be negative."
+        assert (self.backlog == False) | (self.backlog == True), "The backlog parameter must be a boolean."
+        assert m >= 2, "The minimum number of stages is 2. Please try again"
+        assert len(self.unit_cost) == m, "The length of r is not equal to the number of stages."
+        assert len(self.demand_cost) == m, "The length of k is not equal to the number of stages."
+        assert len(self.holding_cost) == m, "The length of h is not equal to the number of stages - 1."
+        assert len(self.supply_capacity) == m-1, "The length of c is not equal to the number of stages - 1."
+        assert len(self.lead_time) == m-1, "The length of L is not equal to the number of stages - 1."
+        assert self.dist in [1,2,3,4,5], "dist must be one of 1, 2, 3, 4, 5."
+        if self.dist < 5:
+            assert distributions[self.dist].cdf(0,**self.dist_param), "Wrong parameters given for distribution."
+        else:
+            assert len(self.user_D) == self.num_periods, "The length of the user specified distribution is not equal to the number of periods."
+        assert (self.alpha>0) & (self.alpha<=1), "alpha must be in the range (0,1]."
+
+        # select distribution
+        self.demand_dist = distributions[self.dist]
+
+        # set random generation seed (unless using user demands)
+        if self.dist < 5:
+            self.seed(self.seed_int)
+
+        # intialize
+        self.reset()
+
+        # action space (reorder quantities for each stage; list)
+        self.pipeline_length = (m-1)*(lt_max+1)
+        self.action_space = gym.spaces.Box(
+            low=np.zeros(m-1), high=self.supply_capacity, dtype=np.int16)
+        # observation space (Inventory position at each echelon, which is any integer value)
+        self.observation_space = gym.spaces.Box(
+            low=-np.ones(self.pipeline_length)*self.supply_capacity.max()*self.num_periods*10,
+            high=np.ones(self.pipeline_length)*self.supply_capacity.max()*self.num_periods, dtype=np.int32)
+
+    def number_of_agents(self) -> int:
+        return 1
+
+    # def action_space(self, agent_id) -> typing.Union[Continuous, Discrete]:
```

Comment:
@maxpumperla I'm trying to convert the OR Gym inventory model to a Pathmind Simulation but I'm confused about the action space setup. I think it's continuous, how should that be set up in this case?

Reply:
Union of Countinuous and Discrete tells you that the return type is either one of them. If you have a Continuous space, you create and return that. If it's discrete, you use discrete.

Reply:
Sorry, I meant to ask if you understand the action space in this particular simulation. 

I read code like a 90-year old so it takes me a long time. I was hoping you would know at a glance to save me a couple of hours. I was just being selfish so no worries if you don't have the bandwidth to look into it.

Reply:
@maxpumperla I think we got it working but we were confused about one thing. I'm using `Discrete(100, 3)` but it should be a tuple of various lengths instead. Like below.

Action 0: n = 100
Action 1: n = 90
Action 2: n = 80

Can simulation.py accommodate that?